### PR TITLE
[Doc] Improvements on conv/pool/fold/stft/ParamDict docs

### PR DESCRIPTION
**Pull request by ssnl**
Also fixes some incorrect formula rendering.

Test plan:

make html

### CODE REVIEW

File: torch/functional.py [hunk 1]
```diff
@@ -286,7 +286,7 @@ def stft(input, n_fft, hop_length=None, win_length=None, window=None,
 
     .. math::
         X[m, \omega] = \sum_{k = 0}^{\text{win\_length}}%
-                            window[k]\ input[m \times hop_length + k]\ %
+                            window[k]\ input[m \times \text{hop\_length} + k]\ %
                             e^{- j \frac{2 \pi \cdot \omega k}{\text{win\_length}}},
```
Comment (weiyangfb) [inline torch/functional.py]: not a big fan of `frac` after `e^` because it is  hard to read text at `frac`, maybe use `exp{}` instead?

File: torch/nn/modules/conv.py [hunk 2]
```diff
@@ -235,13 +236,13 @@ class Conv2d(_ConvNd):
 
     .. note::
 
-         The configuration when `groups == in_channels` and `out_channels == K * in_channels`
+         The configuration when ``groups == in_channels`` and ``out_channels == K * in_channels``,
```
Comment (weiyangfb) [inline torch/nn/modules/conv.py]: also change them at `conv1d`

Comment (weiyangfb) [reply torch/nn/modules/conv.py]: also, it is hard to understand what it mean at first glance. Maybe rephrase to "It is also know as `depwise convolution` when `group == in_channels` and `out_channels == K \star in_channels`, where `K` is a positive integer"

File: torch/nn/modules/conv.py [hunk 3]
```diff
@@ -357,13 +360,13 @@ class Conv3d(_ConvNd):
...
          where `K` is a positive integer is termed in literature as depthwise convolution.
 
          In other words, for an input of size :math:`(N, C_{in}, D_{in}, H_{in}, W_{in})`, if you want a
          depthwise convolution with a depthwise multiplier `K`,
          then you use the constructor arguments
-         :math:`(in\_channels=C_{in}, out\_channels=C_{in} * K, ..., groups=C_{in})`
+         :math:`(\text{in\_channels}=C_{in}, \text{in\_channels}=C_{in} * K, ..., \text{groups}=C_{in})`
```
Comment (weiyangfb) [inline torch/nn/modules/conv.py]: `* K` -> `\times K` ?

File: torch/nn/modules/conv.py [hunk 4]
```diff
@@ -235,13 +236,13 @@ class Conv2d(_ConvNd):
...
 
-         The configuration when `groups == in_channels` and `out_channels == K * in_channels`
+         The configuration when ``groups == in_channels`` and ``out_channels == K * in_channels``,
          where `K` is a positive integer is termed in literature as depthwise convolution.
 
          In other words, for an input of size :math:`(N, C_{in}, H_{in}, W_{in})`, if you want a
          depthwise convolution with a depthwise multiplier `K`,
          then you use the constructor arguments
-         :math:`(in\_channels=C_{in}, out\_channels=C_{in} * K, ..., groups=C_{in})`
+         :math:`(\text{in\_channels}=C_{in}, \text{in\_channels}=C_{in} * K, ..., \text{groups}=C_{in})`
```
Comment (weiyangfb) [inline torch/nn/modules/conv.py]: * K -> \times K ?

File: torch/nn/modules/fold.py [hunk 5]
```diff
@@ -15,11 +15,12 @@ class Fold(Module):
...
     operation combines these local blocks into the large :attr:`output` tensor
-    of shape :math:`(N, C, \text{output\_size}[0], \text{output\_size}[1], \dots)`.
-    Similar to :class:`~torch.nn.Unfold`, the arguments must satisfy
+    of shape :math:`(N, C, \text{output\_size}[0], \text{output\_size}[1], \dots)`
+    by summing the overlapping values. Similar to :class:`~torch.nn.Unfold`, the
+    arguments must satisfy
 
     .. math::
-        L = \prod_d \left\lfloor\frac{\text{output\_size}[d] + 2 \times \text{padding}[d] \
+        L = \prod_d \left\lfloor\frac{\text{output\_size}[d] + 2 \times \text{padding}[d] %
```
Comment (weiyangfb) [inline torch/nn/modules/fold.py]: what does `%` do here?

File: torch/nn/modules/fold.py [hunk 6]
```diff
@@ -59,6 +61,13 @@ class Fold(Module):
     * For the case of two output spatial dimensions this operation is sometimes
       called ``col2im``.
 
+    .. note::
```
Comment (weiyangfb) [inline torch/nn/modules/fold.py]: why do we need to mention `unfold` and `block overlap` here?

File: torch/nn/modules/fold.py [hunk 7]
```diff
@@ -152,6 +161,13 @@ class Unfold(Module):
     * For the case of two input spatial dimensions this operation is sometimes
       called ``im2col``.
 
+    .. note::
```
Comment (weiyangfb) [inline torch/nn/modules/fold.py]: it is fine to mention them here

File: torch/nn/modules/conv.py [hunk 8]
```diff
@@ -414,7 +420,7 @@ class Conv3d(_ConvNd):
 
     .. _link:
         https://github.com/vdumoulin/conv_arithmetic/blob/master/README.md
```
Comment (weiyangfb) [inline torch/nn/modules/conv.py]: these links do not show up correctly

File: torch/nn/modules/fold.py (hunk 5 above)
Comment (SsnL) [reply torch/nn/modules/fold.py]: roughly does line continuation in latex

File: torch/nn/modules/pooling.py [hunk 9]
```diff
@@ -227,29 +226,29 @@ class MaxUnpool1d(_MaxUnpoolNd):
     including the indices of the maximal values and computes a partial inverse
     in which all non-maximal values are set to zero.
 
-    .. note:: `MaxPool1d` can map several input sizes to the same output sizes.
-              Hence, the inversion process can get ambiguous.
+    .. note:: :class:`MaxPool1d` can map several input sizes to the same output
```
Comment (weiyangfb) [inline torch/nn/modules/pooling.py]: `~torch.nn.MaxPool1d`

File: torch/nn/modules/pooling.py [hunk 10]
```diff
@@ -292,32 +291,32 @@ class MaxUnpool2d(_MaxUnpoolNd):
     including the indices of the maximal values and computes a partial inverse
     in which all non-maximal values are set to zero.
 
-    .. note:: `MaxPool2d` can map several input sizes to the same output sizes.
-              Hence, the inversion process can get ambiguous.
+    .. note:: :class:`MaxPool2d` can map several input sizes to the same output
```
Comment (weiyangfb) [inline torch/nn/modules/pooling.py]: `~torch.nn.MaxPool2d`

File: torch/nn/modules/pooling.py [hunk 11]
```diff
@@ -514,14 +513,14 @@ class AvgPool2d(_AvgPoolNd):
 
     Shape:
         - Input: :math:`(N, C, H_{in}, W_{in})`
-        - Output: :math:`(N, C, H_{out}, W_{out})` where
+        - Output: :math:`(N, C, H_{out}, W_{out})`, where
 
           .. math::
-              H_{out} = \left\lfloor\frac{H_{in}  + 2 * \text{padding}[0] -
+              H_{out} = \left\lfloor\frac{H_{in}  + 2 \times \text{padding}[0] -
```
Comment (weiyangfb) [inline torch/nn/modules/pooling.py]: also change `*` -> `\times` above

File: torch/nn/modules/pooling.py (hunk 9 above)
Comment (SsnL) [reply torch/nn/modules/pooling.py]: I don't think this is a problem. We did all the tensor.* docs this way. Moreover, the name will be clickable to redirect to MaxPool1d.

File: torch/nn/modules/pooling.py [hunk 12]
```diff
@@ -559,9 +558,9 @@ class AvgPool3d(_AvgPoolNd):
     .. math::
 
         \text{out}(N_i, C_j, d, h, w)  = \sum_{k=0}^{kD-1} \sum_{m=0}^{kH-1} \sum_{n=0}^{kW-1}
-                \frac{\text{input}(N_i, C_j, \text{stride}[0] * d + k, \text{stride}[1] * h + m,
-                        \text{stride}[2] * w + n)}
-                     {kD * kH * kW}
+                \frac{\text{input}(N_i, C_j, \text{stride}[0] \times d + k, \text{stride}[1] \times h + m,
```
[the remark this answers is not in the document]
Comment (SsnL) [reply torch/nn/modules/pooling.py]: Fixed!

**Comment by weiyangfb** [comment]
Looks good! There are couple places lacking `.. math::` that can use a quick fix: 

https://github.com/pytorch/pytorch/blob/684b55d76257141aa9ea6eb55923896ca4f7d029/torch/nn/modules/conv.py#L260-L265

https://github.com/pytorch/pytorch/blob/684b55d76257141aa9ea6eb55923896ca4f7d029/torch/nn/modules/conv.py#L382-L390

https://github.com/pytorch/pytorch/blob/684b55d76257141aa9ea6eb55923896ca4f7d029/torch/nn/modules/conv.py#L654-L659

https://github.com/pytorch/pytorch/blob/684b55d76257141aa9ea6eb55923896ca4f7d029/torch/nn/modules/conv.py#L794-L802
